pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -155,6 +155,7 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 		podSpec.Containers = []v1.Container{
 			*container,
 		}
+		podSpec.Volumes = isvcutils.MergeRuntimeVolumes(sRuntime.Volumes, isvc.Spec.Predictor.Volumes)
```

Comment:
Would it make sense to have this call happen within `MergePodSpec` since this is a PodSpec field (similar to how `MergeEnvs` is called from within `MergeRuntimeContainers`)?

Reply:
Sure, I think I only added it here to match how `MergeRuntimeContainers` was done. I could move that in there too?

Reply:
Hmm I guess the situation is a bit more complicated since there's a `Containers` list in the inline predictor PodSpec and a separate single `Container` in the `PredictorExtensionSpec`. I assume we're only considering the latter for consistency with how the other types of non-custom predictor work but there isn't really any why you shouldn't be able to add/override multiple containers when using a `ServingRuntime`. The logic right now kind of assumes that there's only one.

Perhaps we should at least look for one in the SR with the same name and merge that - and if the name is different we could just add it rather than merging with the first one in the list?